Modules/_ssl.c

```diff
@@ -5525,8 +5548,7 @@ PySSLSession_richcompare(PyObject *left, PyObject *right, int op)
     } else {
         const unsigned char *left_id, *right_id;
         unsigned int left_len, right_len;
-        left_id = SSL_SESSION_get_id(((PySSLSession *)left)->session,
-                                     &left_len);
+        left_id = SSL_SESSION_get_id(self->session, &left_len);
```

Comment:
I prefer `((PySSLSession *)left)->session` to make the code more symmetrical.

You may use PySSLSession_CAST() on these two lines.